### ISSUE TITLE
Zimlet failing to actually do anything after being installed

### ISSUE THREAD

**Issue by StephanieSunshine** [Zimbra-Community/owncloud-zimlet#45]
Hi there,

I've been using your Zimlet with my server for a few weeks now and everything was working perfectly, but for some reason the integration shows, but doesn't do anything.  I can right click and see preferences, but I get nothing.  My web browser's debug log says:

```
------------------------------------- Loading package: Zimlet
Uncaught SyntaxError: Unexpected string     tk_barrydegraaff_owncloud_zimlet.js?v=141215154045&language=en&country=US&debug=1:89
Uncaught ReferenceError: ownCloudZimlet is not defined(anonymous function) @ mime.js?v=141215154045&language=en&country=US&debug=1:1
------------------------------------- Loading package: Contacts
```
 looking at line 89 I find:
```
html+="<tr><td>Default folder:</td><td><input readonly style='width:98%; background-color:#eeeeee' type='text' id='owncloud_zimlet_default_folder' value='"+decodeURIComponent(this.getUserProperty("owncloud_zimlet_default_folder"))+"'></td></tr>"+"<tr><td></td><td><table><tr><td><input type='checkbox' id='set_new_default'></td><td><small>Set a new default folder.</small></td></tr></table></td></tr>""</table></div>";this._dialog.setContent(html);this._dialog.setButtonListener(DwtDialog.OK_BUTTON,new AjxListener(this,this.prefSaveBtn));this._dialog.setButtonListener(DwtDialog.CANCEL_BUTTON,new AjxListener(this,this.cancelBtn));break;}
```
Which I would assume it's having a problem with the results of this.getUserProperty "owncloud_zimlet_default_folder".  I looked into my at my users via the console to see what that property should look like, and for all of my users it's blank.  I tried for a for while to solve this, but it's eluding me.  I sync'd the git repo just in case and reinstalled per your instructions.  I wasn't missing any files when I pulled the repo.  The only possible thing I can think of is we enabled caching with Cloudflare, but turned off things like the rocket loader so our js files didn't get messed up.

Everything else seems to be in perfect working order.  Do you have any idea what might be causing this?

**Comment by barrydegraaff** [comment]
It sounds like a weird cache issue.

1. Flush your browser cache, remove all temporary internet files
2. On the server zmprov fc all
3. On the server zmcontrol restart

Should do.

Take a look at dmesg and syslog, see if something weird is happening to your storage.

**Comment by barrydegraaff** [comment]
Also, it takes some tome for Zimbra to load... even when it looks ready, it may not be.

So wait for 2 minutes after loading the web interface and try then (or wait for the mini calendar to show).

It may wel be that the Zimlet is not yet loaded

**Comment by StephanieSunshine** [comment]
First off, I don't understand why you closed this ticket.  The server is fine, every vm on it is perfectly ok and not having any sort of disk errors ( I did look ).  The server sits on a 100mbit line not 5 miles from me and my 200mbit line with a 13ms ping time.

The calendar comes up instantly.  Just to make things simple, I logged in to my email with Firefox this time instead of Chrome, which has never been connected to Zimbra and I'm almost the exact same problem in console.  I even made sure to clear my cache 100% in Firefox before attempting.

Even better, Firefox gave a me a little bit better of a hint saying a semicolon was missing at line 89, col 401.  Deminifying the line shows a semicolon is indeed missing.

```
html += "<tr><td>Default folder:</td><td><input readonly style='width:98%; background-color:#eeeeee' type='text' id='owncloud_zimlet_default_folder' value='" + decodeURIComponent(this.getUserProperty("owncloud_zimlet_default_folder")) + "'></td></tr>" + "<tr><td></td><td><table><tr><td><input type='checkbox' id='set_new_default'></td><td><small>Set a new default folder.</small></td></tr></table></td></tr>"
"</table></div>";
this._dialog.setContent(html);
this._dialog.setButtonListener(DwtDialog.OK_BUTTON, new AjxListener(this, this.prefSaveBtn));
this._dialog.setButtonListener(DwtDialog.CANCEL_BUTTON, new AjxListener(this, this.cancelBtn));
break;
}
```

**Comment by StephanieSunshine** [comment]
Oh and I did flush the cache and restart all the services on the mail server like you asked before I attempted this again.

**Comment by barrydegraaff** [comment]
The ticket is closed as I cannot reproduce it, as soon as I can reproduce I will re-open.

In tested the code here, and it works as usual. I tested with existing and new accounts on Google Chrome.

The line number does not seem to match with the code you are describing as on line 89 is:
https://github.com/Zimbra-Community/owncloud-zimlet/blob/master/tk_barrydegraaff_owncloud_zimlet/tk_barrydegraaff_owncloud_zimlet.js#L89

The closing ; is on line 90... I will look again tomorrow, but as you said, it did work before for you for a couple of weeks, so if it is not working now, *something* has changed, that could be a browser update, but then I would also see that by now... so it has to be something else, but I don't know what that may be.

**Comment by StephanieSunshine** [comment]
I'm happy to give you a login to the portal if that would help you at all.

**Comment by barrydegraaff** [comment]
You can send account details for zcs/owncloud and an administrative account for your server to info@barrydegraaff.tk.

Public Key Server -- Get "0x725ddd156d36a2d0 "

-----BEGIN PGP PUBLIC KEY BLOCK-----
Version: SKS 1.1.5
Comment: Hostname: keyserver.ubuntu.com

mQGiBEVI10kRBADsCyAEtntq0Mt7RNjpFIEAOFOBZ7yZMue4UYJKfUc39jliZUyQ262hQW3L
X9AjwcEdJSclMmvfD/Z9XebeKCCmmqvHoU+XuKQ69RG+693ukiCKvmo5niCWHzPpgCcPRO9k
sDyPSJR2VY3/43bBiwFOxQa4MRuEyy8Gjn1shUu7hwCgh4oUG+0066CjQaW1OkMT5OrIOh0D
/272R/Ls/EfWSx+jjT3d0YTHp9z79RbXlkv92166r1GSrK8Wvky5S5JnbgCDpThVvThx4COt
TN76PnrQ92vE1QCnPlGmUcErDO86y2EVUlSY2AUm0QAcf7dmYs8UTmGm3WO9RFi1Q0njmW2h
PMSXkelHom4o0Y8wXryge8sXylx8A/9BlV+hBRfMJVxbJLryphZqRnRZcNY/gd6dOyqQzoSM
Nb01frug2HsnLdFi2LLqevs5sTngjPdpnEh69CHb6aJpcmcfupxiOnj+nscRKa5WkI2vZNXb
Bq8s9epgGyCwyGozgAISv+WiM0UZUvUpcKHT1PrqX3yClku4S1MNTFuHhbQ+QmFycnkgZGUg
R3JhYWZmICh3d3cuYmFycnlkZWdyYWFmZi50aykgPGluZm9AYmFycnlkZWdyYWFmZi50az6I
YAQTEQIAIAUCRUjXSQIbAwYLCQgHAwIEFQIIAwQWAgMBAh4BAheAAAoJEHJd3RVtNqLQsScA
n3GfVLXodd36vc2oZ6ZTc3OcYk0fAJ4q1yAvUfGCJywErBMYxve1wcRsSrkEDQRFSNhDEBAA
4SuDfG3chpcGY749mKEvlF9XIt87tTPr5oeHg1xI0vwyXvqGCQPtyRwL3+Q4ki7K6Mi3njMg
tT8wCiJGf57HRcndCSolfXcfFhp0jZQO3Jn5CyYHDTWjAGPHuHDDmh04j+9SRsdKmjjan5Xj
AUYknXevKSNbLxuUw1UV6Djznq73MzAtnh9VC6KsqYXwpXBCJkc9QJXeSUGxNfoooTZ1DOno
Pm8LeDwWmzSk+A8PkVP7Q3NINK0WJyBNg+XVTp/J8RfeFbavNUKXdIgF7nKbEevuDuo++w09
D4Lc86wnWR47AaXReJwMn0URvJN5TW+3UkL+nhr0vBOVfIKVn4c6OpArzyaE6tVZ+zoCoTsT
RAUq5/h+of5rvoGFiN7pJzJaYHKPrsYdaQ+d1SDLR8/UNiZwB38N2MaqawGDjIjWWl21+mr4
eq7AZCMVM1HiffGP9FFkRv0MOhoupM5PN9VYG4B1cA+CFKKx+so88RXIU4hmW5D/YZHdK8YR
NrqyTSC2MlKuXXuxVrIQXoCeXFKl1XK4LREk3RU+fWE+sYlgmiRefAc8ttUeKZXVef7v2LTS
bSdW7BG46S9oC8+xKtFwyGZUKU7yzmYW8+LLNt8Y4Q9VYUgMqm8pYwAAjYcb9S9gUhbwVbZd
BheyNrXtOpS0TdUgEzpCb8eUVYC1zh3JYHcABAsP/j6f/Lk/6G9Kk9jzeAKGVgU/1FuHKe/M
qpQdYWuzj8wVRyvwZVWPHHgsZIDxz1Nplc1An4G16GCSADmyIuXQtBoGwtFxSf7ZGZjnHG/F
mtTQFSo0XvasOf9b998We5HzjGOh+2xn4tod7qBEDUjIR5AQhYf3Crzxs40tynudsjm4ZL4W
TkhG959PPjBZjyaur2Ox2m8nNhGuh8gC2Ypg9LxUPPR+5VKS1Y2dxtem0y5MaGKGpV+PhOzA
KFgQr58p52f1rrV/8aV+9aPLPr3b0yDpqL+FUl57otb3dgV/S4v36Fv0c4UquOvuhpoZO0cD
CDTRmFQzrGmOF3tCEOD/V1NC4K2HPI1OWllF12dzt1z7dq1UOD78dqrpRCz8zJL2D5jqW1UY
PY1HDbzqOvhXWuQssIl5B4ILbMIkuvfTOvKFRoXgEVyT3+cwFZZHu3hkBWWWzbli7Sx/IQIu
FrA9/9fjM6SgAVhwrzTUAqNcyjzwnf+pCORtPzJ8DOro4JB7kPqIxP2O506H49V4MDk9zbAK
Nh/Neq7eZHUk0LeDWyJ4DREfVfMtlGiukun+Kg0iu72MKy9O7NA10RgcsXWf4U2+8ojAACP+
vhRpAQ8oIggmIJ/V2iO1ke3v5iWOnu85QkWblCACXuqVGa5mHJFEKflFlm9bNlTN7qDq6QXa
VeMhiEkEGBECAAkFAkVI2EMCGwwACgkQcl3dFW02otBeqACeKvTDlFewtO0PKz06wmJdr3mf
mVUAnRLWaGMwRaR1bPZJGgvNBMRxxT/m
=gTSf
-----END PGP PUBLIC KEY BLOCK-----

**Comment by StephanieSunshine** [comment]
Ok so for the record, a few things were causing problems.  The big one was cloudflare was trying to mask my SSL certificate because when I went from caching to not, the hostname on the cert no longer matching and I used an A record instead of a CNAME to create an accelerated variant of the webclient.  This made the server hostname not match what the client was seeing, but because Cloudflare had their own cert in play, it was masking that little detail from me.  I assume if I went into Zimbra and told it the portal was at a different address and updated the certificate that it would work, but frankly, I found the WAF like stuff for Zimbra this morning while I was trying to undo some of the layers and I don't see any real speed gain compared to the server itself when using cloudflare to offload.

Things are working now, my best advice I have for anyone that might possibly trip over this in the future trying to solve a problem is this,  Cloudflares rocket loader is a real pita to get to play right and when hostnames don't exactly align ( I needed mail.* for the mx / imaps / smtps stuff to be exposed for the mobile devices, so it should have been a CNAME to get back to mail.* from webmail.* but with the workload of this week, I had turned it into an A record. ugh ) Zimbra will work for the most part, but it does get confused in places and it's best not to try and confuse Zimbra.
